src/main/java/org/gridsuite/notification/server/NotificationWebSocketHandler.java

```diff
@@ -110,8 +115,15 @@ public NotificationWebSocketHandler(ObjectMapper jacksonObjectMapper, @Value("${
     @Override
     public Mono<Void> handle(WebSocketSession webSocketSession) {
         URI uri = webSocketSession.getHandshakeInfo().getUri();
-        MultiValueMap<String, String> parameters = UriComponentsBuilder.fromUri(uri).build().getQueryParams();
+        MultiValueMap<String, String> parameters = UriComponentsBuilder.fromUri(uri).build(true).getQueryParams();
         String filterStudyName = parameters.getFirst(QUERY_STUDY_NAME);
+        if (filterStudyName != null) {
+            try {
+                filterStudyName = java.net.URLDecoder.decode(filterStudyName, StandardCharsets.UTF_8.name());
```

Comment:
Why are you using fully qualified name?

Reply:
What do you mean with "fully qualified" ?

Reply:
You mean this "java.net.URLDecoder" ?